cache/cache.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2016 Mesosphere, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cache is a simple, local, in-memory key-value store.
+package cache
+
+import "sync"
+
+// Object represents a single object in the cache. Although this could be represented
+// as a bare interface{}, we chose to create a new Object type in case we wish to add
+// additional functionality or metadata in the future.
+type Object struct {
+	Contents interface{}
+}
+
+// Cache represents the structure of the objects in the cache (objects)
+// and a single locking mechanism (mutex) that is shared across a given instance
+// of the cache.
+type Cache struct {
```

Comment:
would rather see `Cache` as an exported `interface` type and `cachImpl` or something as an internal struct. Then you could build additional implementations in the future: time-based cache, size-based cache, etc. that all conform to the same basic `Cache` interface (which should be as narrow as possible).